students/erik_oien/lesson05/mailroom3.py

```diff
@@ -30,42 +30,33 @@ def thank_you():
             sub_menu_switch[donor_name.lower()]()
         elif donor_name.lower() == 'list':
             past_donor = False
-            for donor in donor_dict.items():
-                print(donor[1]["first_name"] + " " + donor[1]["last_name"])
+            for donor in donor_dict.keys():
+                print(donor)
         else:
-            try:
-                for donor in donor_dict.items():
-                    if donor_name.lower() == (donor[1]["first_name"].lower() + " " + donor[1]["last_name"].lower()).strip():
-                        past_donor = True
-                        donation_amount = input("How much was the most recent donation from {}? >>> ".format(donor_name))
-                        donor[1]["donations"].append(int(donation_amount))
-                if past_donor == False:
-                    print("Looks like this is a new donor, lets add them!")
-                    first_name, last_name = donor_name.split(" ", 1)
-                    donation_amount = input("How much did {} donate? >>> ".format(donor_name))
-                    new_donor = {"first_name": first_name, "last_name": last_name, "donations": [int(donation_amount)]}
-                    donor_dict[f"donor{len(donor_dict)+1}"] = new_donor
-                email = thank_you_email(donor_name, donation_amount)
-                print(email)
-                return False
-            except(ValueError):
-                print("Expecting first and last name, only one name entered")
-                
+            for donor in donor_dict:
+                if donor_name.lower() == donor.lower():
+                    past_donor = True
+                    donation_amount = input("How much was the most recent donation from {}? >>> ".format(donor_name))
+                    donor_dict[donor].append(int(donation_amount))
+            if past_donor == False:
+                print("Looks like this is a new donor, lets add them!")
+                donation_amount = input("How much did {} donate? >>> ".format(donor_name))
+                donor_dict[donor_name] = [int(donation_amount)]
```

Comment:
you'll need to update this section as well. 

to search for a donor in a dictionary, utilize dict efficient look up:
```
if donor_dict.get(donor):
    # update donations
else:
    # add new donor.
```